File: packages/js/core/src/algorithms/uri-resolution/resolvers/RedirectsResolver.ts
import { applyRedirects } from "../..";
import { Uri, Client } from "../../..";
import { UriResolutionResult } from "./UriResolutionResult";
import { UriToApiResolver } from "./UriToApiResolver";

export class RedirectsResolver implements UriToApiResolver {
  name = "Redirect";

  async resolveUri(uri: Uri, client: Client): Promise<UriResolutionResult> {
    let redirectedUri = applyRedirects(uri, client.getRedirects({}));
    
    return Promise.resolve({
      uri: redirectedUri,
    });
  };
}

File: packages/py/core/src/uri_resolution/resolvers/redirects/redirects_resolver.py
from __future__ import annotations
from typing import Awaitable

from ....algorithms import apply_redirects
from ...core import UriResolver, UriResolutionResult
from .... import GetRedirectsOptions


class RedirectsResolver(UriResolver):
    @property
    def name(self) -> str:
        return type(self).__name__

    async def resolve_uri(
        self, uri: Uri, client: Client, cache: ApiCache, resolution_path: UriResolutionStack
    ) -> Awaitable(UriResolutionResult):
        redirect_uri = apply_redirects(uri, client.get_redirects(GetRedirectsOptions("")))

        return UriResolutionResult(uri=redirect_uri)
